### PR TITLE
Create the first version of a parser for empik.com.

### PR DESCRIPTION
**Pull request by tdudzik**
The process of parsing empik.com is following:
1. First we need to extract urls for all categories from http://www.empik.com/ebooki/promocje
2. Then, from every page of category we need to extract urls for every book
3. And finally we need to extract book data from page of single book

I've also used parallel streams for better performance. It wasn't precisely measured but in some cases parsing was ~ 3x faster than using normal streams.

### CODE REVIEW

File: RobotCrawler/src/main/java/parser/empik/EmpikUrlResolver.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+class EmpikUrlResolver {
+
+    static String resolveUrl(String url) {
+        URI empikUri = null;
+        try {
+            empikUri = new URI("http://empik.com/");
+        } catch (URISyntaxException e) {
+            e.printStackTrace();
+        }
+        return empikUri.resolve(url).toString();
```
Comment (LIttleAncientForestKami) [inline RobotCrawler/src/main/java/parser/empik/EmpikUrlResolver.java]: Possible NPE :skull: :-1:

**Comment by LIttleAncientForestKami** [comment]
Nice and pretty code. Few things to improve though.

1) Noticed your readme sends folks away to Wiki which are... empty.
2) No mention of Lombok. Your code doesn't compile in my IDE (only after poking around I realized why and could fix). 
3) BookData is immutable? And it's Javadoc starts with _Class responsible for_ - :recycle: - things like "this class" in **class** Javadoc deserve a :warning: . More remarks placed on specific lines.

Big :clap: for lambdas, streams and overall code cleanliness.

File: RobotCrawler/src/main/java/parser/empik/EmpikUrlResolver.java (hunk 1 above)
Comment (tdudzik) [reply RobotCrawler/src/main/java/parser/empik/EmpikUrlResolver.java]: I've improved that in this commit: https://github.com/team-javapocalypse-ja/Robot/pull/6/commits/4de45ff2b508f0fe7665e9da1988dc13c10e78e8

**Comment by LukaszCzubernat** [comment]
Nice work :smile: 
For me you can marge, but wait for review from other team

File: RobotCrawler/src/main/java/parser/empik/PageDownloader.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+    public Optional<Document> downloadPage(String url) {
+        Document document;
+        try {
+            document = Jsoup.connect(url).get();
+        } catch (IOException e) {
+            document = null;
+        }
+
+        return Optional.ofNullable(document);
```
Comment (caroo7) [inline RobotCrawler/src/main/java/parser/empik/PageDownloader.java:19]: :+1:

**Comment by caroo7** [comment]
:ok:

**Pull request merged**